apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/dot-template-create-edit.component.ts

```diff
@@ -75,6 +74,30 @@ export class DotTemplateCreateEditComponent implements OnInit, OnDestroy {
         });
     }
 
+    /**
+     * Update Working Tempalte
+     *
+     * @memberof DotTemplateCreateEditComponent
+     */
+     updateTemplate({ layout, body, themeId }: DotTemplate): void {
+        let value = {
+            ...this.form.value,
+            body
+        };
+        if (layout) {
+            value = {
+                ...this.form.value,
+                layout,
+                theme: themeId
+            };
+
+            // Design templates need to be save 10 seconds after the last change.
+            this.store.saveTemplateDebounce(value);
```

Comment:
Why don't do this inside the store? for example, in `updateWorkingTemplate` since we know if is a design template there and they way we let all the logic in the store.